kyaml/kio/kio.go

```diff
@@ -112,6 +108,9 @@ func (p Pipeline) ExecuteWithCallback(callback PipelineExecuteCallbackFunc) erro
 			callback(op)
 		}
 		result, err = op.Filter(result)
+		// TODO: This len(result) == 0 should be removed and empty result list should be
+		// handled by outputs. However currently the some writer like LocalPackageReadWriter
+		// will clear the output directory and which will cause unpredictable results
 		if len(result) == 0 || err != nil {
```

Comment:
@monopole @prachirp This if cannot be removed now.

Reply:
Please file an issue for this, and in the TODO note the issue number.